overlord/servicestate/servicestate.go

```diff
@@ -397,3 +399,26 @@ func SnapServiceOptions(st *state.State, instanceName string, quotaGroups map[st
 
 	return opts, nil
 }
+
+// SnapAppsLogReader returns an io.ReadCloser which produce logs for the provided
+// snap AppInfo's. It is a convenience wrapper around the systemd.LogReader
+// implementation.
+func SnapAppsLogReader(appInfos []*snap.AppInfo, n int, follow bool) (io.ReadCloser, error) {
+	serviceNames := make([]string, len(appInfos))
+	for i, appInfo := range appInfos {
```

Comment:
now that this is an exported helper, this needs to either skip non-service appInfos or return an internal error on them